src/DefaultBuilder/src/StaticWebAssets/StaticWebAssetsWebHostBuilderExtensions.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+
+namespace Microsoft.AspNetCore
+{
+    /// <summary>
+    /// Extensions for configuring static web assets for development.
+    /// </summary>
+    public static class StaticWebAssetsWebHostBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the <see cref="IWebHostEnvironment.WebRootFileProvider"/> to use static web assets
+        /// defined by referenced projects and packages.
+        /// </summary>
+        /// <param name="builder">The <see cref="IWebHostBuilder"/>.</param>
+        /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+        public static IWebHostBuilder UseStaticWebAssets(this IWebHostBuilder builder)
+        {
+            builder.ConfigureAppConfiguration((context, configBuilder) =>
```

Comment:
We do it in ConfigureAppConfiguration to avoid adding a new lifecycle method in hosting. This happens early enough in the configuration process that makes it really hard to observe the wrong WebRootFileProvider

Reply:
@Tratcher @davidfowl - if we need to replace the file provider, what's the best way to do that?

Reply:
We need to add something, there's nothing for this today.

Reply:
@davidfowl I kept it minimalistic because I wanted your input on this. Originally I had a ConfigureWebHostEnvironment addition to CreateAppConfiguration, but I didn't find it justified enough as the only thing to configure there are the file providers.

I also didn't want to invent anything based on settings or similar nor bake any concept into hosting, as it would be hard to make something generic based on settings for what we need.

Reply:
ConfigureAppConfiguration is your best option for now. StaticWebAssets are cumulative so it's unlikely to break anyone else. Also, I've seen very few examples of anyone setting WebRootFileProvider so it's unlikely anyone else will break you.

Reply:
Yeah, I don't think this will break people or that people will be breaking this by changing it. It will be documented, so if someone runs into this, they can simply account for it.